diffrax/_custom_types.py

```diff
@@ -46,6 +46,7 @@
 Y = PyTree[Shaped[ArrayLike, "?*y"], "Y"]
 VF = PyTree[Shaped[ArrayLike, "?*vf"], "VF"]
 Control = PyTree[Shaped[ArrayLike, "?*control"], "C"]
+_Control = TypeVar("_Control", bound=Control)
```

Comment:
On which note, I'd suggest not putting `_Control` here, but just copying it into every file it's needed. This (a) means that the `bound` can be refined, and (b) I generally try not to import `TypeVar`s, but instead treat them as part of the implementation of whatever it is they're subcripting.

(Also (c) the convention in Diffrax is that having an underscore at the start of a name indicates that it is private to the *file* it is in, not to the overall package.)